### PR TITLE
CRD defs for Apache HTTPD Autoinstrumentation

### PR DESCRIPTION
**Pull request by chrlic**
This is a followup PR in Apache HTTPD Autoinstrumentation implementation project to previous PR #1236.

It contains CRD definition, defaults, validations, and test.

Next PR will contain the autoinstrumentation logic itself and e2e tests.

### CODE REVIEW

**Comment by chrlic** [comment]
Updated with generated resources (make generate).

**Comment by chrlic** [comment]
Generated bundle for Apache Httpd CRD (make bundle).

**Comment by chrlic** [comment]
Version updated for make bundle

**Comment by chrlic** [comment]
Chngnd repo user to open-telemetry for make bundle

File: apis/v1alpha1/instrumentation_types.go [hunk 1]
```diff
@@ -145,6 +149,32 @@ type DotNet struct {
...
+	// Env defines Apache HTTPD specific env vars. There are four layers for env vars' definitions and
+	// the precedence order is: `original container env vars` > `language specific env vars` > `common env vars` > `instrument spec configs' vars`.
+	// If the former var had been defined, then the other vars would be ignored.
+	// +optional
+	Env []corev1.EnvVar `json:"env,omitempty"`
+
+	// Attrs defines Apache HTTPD agent specific attributes. The precedence is:
+	// `agent default attributes` > `instrument spec attributes`
+	// +optional
+	Attrs []corev1.EnvVar `json:"attrs,omitempty"`
```
Comment (pavolloffay) [inline apis/v1alpha1/instrumentation_types.go:166]: what exactly is this configuring?

File: config/manager/kustomization.yaml [hunk 2]
```diff
@@ -1,2 +1,8 @@
 resources:
 - manager.yaml
+apiVersion: kustomize.config.k8s.io/v1beta1
```
Comment (pavolloffay) [inline config/manager/kustomization.yaml:3]: this should be reverted

File: apis/v1alpha1/instrumentation_types.go (hunk 1 above)
Comment (chrlic) [reply apis/v1alpha1/instrumentation_types.go:166]: The OTel module for Apache httpd has number of configurable attributes (listed at https://github.com/open-telemetry/opentelemetry-cpp-contrib/tree/main/instrumentation/otel-webserver-module in Configuration section). There are reasonable defaults and some will be set by the instrumentation process, this gives a means to override those defaults.

File: config/manager/kustomization.yaml (hunk 2 above)
Comment (chrlic) [reply config/manager/kustomization.yaml:3]: ok, will do.

Comment (chrlic) [reply config/manager/kustomization.yaml]: Done

**Comment by frzifus** [comment]
linting failed. its fixed in: https://github.com/open-telemetry/opentelemetry-operator/pull/1206/commits/c5de16ab0c1bf0f6d2e11972cec959a7bc2c0455

File: apis/v1alpha1/instrumentation_types.go (hunk 1 above)
Comment (pavolloffay) [reply apis/v1alpha1/instrumentation_types.go:166]: Should we add this link to the godoc?

File: apis/v1alpha1/instrumentation_types.go [hunk 3]
```diff
@@ -145,6 +149,32 @@ type DotNet struct {
...
+
+	// Apache HTTPD server version. One of 2.4 or 2.2. Default is 2.4
+	// +optional
+	Version string `json:"version,omitempty"`
```
Comment (pavolloffay) [inline apis/v1alpha1/instrumentation_types.go:170]: for my curiosity how is this version used?

Comment (chrlic) [reply apis/v1alpha1/instrumentation_types.go:170]: Apache HTTPD used these days is either version 2.2 or (most of the time) 2.4. The OTel agent is a set of binary loadable modules and those are compiled against specific Apache HTTPD version - different headers etc. Therefore you have to use the specific version of agent.

Comment (chrlic) [reply apis/v1alpha1/instrumentation_types.go:170]: Based on the version setting, auto-instrumentation configures Apache HTTPD with the proper set of modules to be loaded. The docker image with the agent contains modules for both supported versions.

File: apis/v1alpha1/instrumentation_types.go (hunk 1 above)
Comment (chrlic) [reply apis/v1alpha1/instrumentation_types.go:166]: Yes, good idea.

Comment (chrlic) [reply apis/v1alpha1/instrumentation_types.go:167]: Done

**Comment by chrlic** [comment]
Hi Pavel, 
could you please have a look at this?
BR,
Martin

**Comment by chrlic** [comment]
Hi Pavol,
could you please have a look at this?
BR,
Martin

File: apis/v1alpha1/instrumentation_types.go (hunk 3 above)
Comment (chrlic) [reply apis/v1alpha1/instrumentation_types.go:171]: Pavol, 
could we move this forward, please?

**Review by pavolloffay — APPROVED** [review]

File: apis/v1alpha1/instrumentation_types.go (hunk 3 above)
Comment (pavolloffay) [reply apis/v1alpha1/instrumentation_types.go:171]: merged, thanks ❤️

**Pull request merged**